features/base/participants/reducer.js

```diff
@@ -131,6 +133,18 @@ ReducerRegistry.register('features/base/participants', (state = [], action) => {
     case PARTICIPANT_LEFT:
         return state.filter(p => p.id !== action.participant.id);
 
+    /**
+     * Remove all participants when connection is disconnected.
+     */
+    case CONNECTION_DISCONNECTED:
+        return [];
```

Comment:
Well, CONFERENCE_LEFT removes the remote participants so, in my opinion, CONNECTION_DISCONNECTED shouldn't have the same task. If you'd defined a sequence of conference lifecycle events, then you would've known (in the example choice that I described in my previous comment) that CONNECTION_DISCONNECTED will have to take care of the local participant only because COFERENCE_LEFT would take care of the remote participants.

Implicitly you've ended with take care of the remote participants in two pieces of code. If I have to do a cleanup per remote participant tomorrow (by invoking a function, for example), I'd have to add that source code in two places. That's a burden to maintenance and increases the risks of introducing bugs.

That's another example for me why a sequence of conference lifecycle events is necessary and it shows how such a sequence improves the application.